src/features/Sync/SyncSlice.ts

```diff
@@ -1,105 +1,156 @@
     password: '',
 };
 
-export interface UpdateHostActionPayload {
-    host: string;
+export interface UpdateLoginUrlActionPayload {
+    loginUrl: string;
 }
-
-export interface UpdateHostAction {
+export interface UpdateLoginUrlAction {
     type: string;
-    payload: UpdateHostActionPayload;
+    payload: UpdateLoginUrlActionPayload;
 }
 
-export interface UpdatePortActionPayload {
-    port: string;
+export interface UpdateSensorUrlActionPayload {
+    sensorUrl: string;
 }
-export interface UpdatePortAction {
+export interface UpdateSensorUrlAction {
     type: string;
-    payload: UpdatePortActionPayload;
+    payload: UpdateSensorUrlActionPayload;
 }
 
-export interface UpdateLoginPathActionPayload {
-    loginPath: string;
+export interface UpdateTemperatureLogUrlActionPayload {
+    temperatureLogUrl: string;
 }
-export interface UpdateLoginPathAction {
+export interface UpdateTemperatureLogUrlAction {
     type: string;
-    payload: UpdateLoginPathActionPayload;
+    payload: UpdateTemperatureLogUrlActionPayload;
 }
 
-export interface UpdateSensorPathActionPayload {
-    sensorPath: string;
+export interface UpdateTemperatureBreachUrlActionPayload {
+    temperatureBreachUrl: string;
 }
-export interface UpdateSensorPathAction {
+export interface UpdateTemperatureBreachUrlAction {
     type: string;
-    payload: UpdateSensorPathActionPayload;
+    payload: UpdateTemperatureBreachUrlActionPayload;
 }
 
-export interface UpdateTemperatureLogPathActionPayload {
-    temperatureLogPath: string;
+export interface UpdateUsernameActionPayload {
+    username: string;
 }
-export interface UpdateTemperatureLogPathAction {
+export interface UpdateUsernameAction {
     type: string;
-    payload: UpdateTemperatureLogPathActionPayload;
+    payload: UpdateUsernameActionPayload;
 }
 
-export interface UpdateTemperatureBreachPathActionPayload {
-    temperatureBreachPath: string;
+export interface UpdatePasswordActionPayload {
+    password: string;
 }
-export interface UpdateTemperatureBreachPathAction {
+
+export interface UpdatePasswordAction {
     type: string;
-    payload: UpdateTemperatureBreachPathActionPayload;
+    payload: UpdatePasswordActionPayload;
 }
 
-export interface UpdateUsernameActionPayload {
+export interface AuthenticateActionPayload {
+    loginUrl: string;
     username: string;
+    password: string;
 }
-export interface UpdateUsernameAction {
+
+export interface  AuthenticateAction {
     type: string;
-    payload: UpdateUsernameActionPayload;
+    payload: AuthenticateActionPayload;
+ } 
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface AuthenticateSuccessActionPayload {}
+
+export interface AuthenticateSuccessAction {
+    type: string;
+    payload: AuthenticateSuccessActionPayload;
 }
 
-export interface UpdatePasswordActionPayload {
-    password: string;
+export interface  SyncSensorsActionPayload {
+    sensorUrl: string;
+ } 
+
+ export interface  SyncSensorsAction {
```

Comment:
Do you think we should do a `types` file in each of the feature dirs? Or is it good in the same file?

Reply:
As discussed, agree good idea to extract into dedicated type definition files.